GitCommands/Remote/GitRemoteManager.cs

```diff
@@ -68,20 +68,27 @@ public GitRemoteManager(Func<IGitModule> getModule)
         [SuppressMessage("ReSharper", "RedundantArgumentDefaultValue")]
         public void ConfigureRemotes(string remoteName)
         {
-            var module = GetModule();
-            var localConfig = module.LocalConfigFile;
+            var remoteNameToUpper = remoteName.ToUpperInvariant();
```

Comment:
Throughout, replace:

- `s.ToUpperInvariant().Contains(other.ToUpperInvariant())` with
- `s.IndexOf(other, StringComparison.InvariantCultureIgnoreCase) != -1`

The latter doesn't allocate temporary strings.

If you prefer, add an extension method to `StringExtensions` that allows `s.Contains(other, StringComparison.InvariantCultureIgnoreCase)` which then calls `IndexOf` internally. If you do this, add a test in `StringExtensionsTests`.